projects/ad7768-evb/src/ad7768_evb.c

```diff
@@ -143,45 +144,191 @@ uint8_t ad7768_evb_verify_status(gpio_device *gpio_dev)
+			.mode = SPI_MODE_0,
+			.platform_ops = &xil_platform_ops,
+			.extra = &xil_spi_initial
+		},
+		/* GPIO */
+		.gpio_reset = {
+			.number = GPIO_RESET_N,
+			.platform_ops = &xil_gpio_platform_ops,
+			.extra = &xil_gpio_initial
+		},
+		.gpio_mode0 = {
+			.number = GPIO_MODE_0_GPIO_0,
+			.platform_ops = &xil_gpio_platform_ops,
+			.extra = &xil_gpio_initial
+		},
+		.gpio_mode1 = {
+			.number = GPIO_MODE_1_GPIO_1,
+			.platform_ops = &xil_gpio_platform_ops,
+			.extra = &xil_gpio_initial
+		},
+		.gpio_mode2 = {
+			.number = GPIO_MODE_2_GPIO_2,
+			.platform_ops = &xil_gpio_platform_ops,
+			.extra = &xil_gpio_initial
+		},
+		.gpio_mode3 = {
+			.number = GPIO_MODE_3_GPIO_3,
+			.platform_ops = &xil_gpio_platform_ops,
+			.extra = &xil_gpio_initial
+		},
+		.gpio_reset_value = GPIO_HIGH,
+		/* Configuration */
+		.pin_spi_input_value = GPIO_HIGH,
+		.sleep_mode = AD7768_ACTIVE,
+		.power_mode = AD7768_FAST,
+		.mclk_div = AD7768_MCLK_DIV_4,
+		.dclk_div = AD7768_DCLK_DIV_1,
+		.conv_op = AD7768_STANDARD_CONV,
+		.crc_sel = AD7768_CRC_16
+	};
+	struct axi_adc *axi_adc_core_desc;
+	struct axi_adc_init axi_adc_initial = {
+		.base = ADC_DDR_BASEADDR,
+		.name = "ad7768_axi_adc",
+		.num_channels = 8
+	};
+	int32_t *data_ptr, i, data_val;
+
+	temp_init.number = GPIO_UP_SSHOT;
+	ret = gpio_get(&temp_desc, &temp_init);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_direction_output(temp_desc, GPIO_LOW);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_remove(temp_desc);
+	if (ret != SUCCESS)
+		return FAILURE;
 
 	/* Configure HDL */
-	gpio_set_direction(&dev->gpio_dev, GPIO_UP_FORMAT_1, GPIO_OUT);
-	gpio_set_value(&dev->gpio_dev, GPIO_UP_FORMAT_1, 0);
-	gpio_set_direction(&dev->gpio_dev, GPIO_UP_FORMAT_0, GPIO_OUT);
-	gpio_set_value(&dev->gpio_dev, GPIO_UP_FORMAT_0, 0);
-	gpio_set_direction(&dev->gpio_dev, GPIO_UP_CRC_ENABLE, GPIO_OUT);
-	gpio_set_value(&dev->gpio_dev, GPIO_UP_CRC_ENABLE,
-			dev->crc_sel ? 1 : 0);
-	gpio_set_direction(&dev->gpio_dev, GPIO_UP_CRC_4_OR_16_N, GPIO_OUT);
-	gpio_set_value(&dev->gpio_dev, GPIO_UP_CRC_4_OR_16_N,
-			dev->crc_sel == AD7768_CRC_4 ? 1 : 0);
-
-	gpio_dev.type = AXI_GPIO;
-	gpio_dev.device_id = XPAR_AD7768_GPIO_DEVICE_ID;
-	gpio_init(&gpio_dev);
-
-	ad7768_evb_clear_status(&gpio_dev);
-	if (ad7768_evb_verify_status(&gpio_dev))
+	temp_init.number = GPIO_UP_FORMAT_1;
+	ret = gpio_get(&temp_desc, &temp_init);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_direction_output(temp_desc, GPIO_LOW);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_remove(temp_desc);
+	if (ret != SUCCESS)
+		return FAILURE;
+
+	temp_init.number = GPIO_UP_FORMAT_0;
+	ret = gpio_get(&temp_desc, &temp_init);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_direction_output(temp_desc, GPIO_LOW);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_remove(temp_desc);
+	if (ret != SUCCESS)
+		return FAILURE;
+
+	temp_init.number = GPIO_UP_CRC_ENABLE;
+	ret = gpio_get(&temp_desc, &temp_init);
```

Comment:
It seems that gpio_get + gpio_direction_output + gpio_remove can be in one function. This way it will be easier to follow the logic of the program.

Reply:
Will place these in a method.